service.smartthings/service.py

```diff
@@ -0,0 +1,334 @@
+            try:
+                if ALL:
+                    events.append(event[2])
+                else:
+                    device = (([x for x in deviceLST if x in event[2]])[0])
+                    if event[2].startswith(device):
+                        if LAST: return event[2]
+                        else: events.append(event[2])
+            except: pass
+        return events
+
+           
+    def checkEvents(self):
+        try:
+            event = (self.getEvents(self.deviceLST) or None)
+            if event != self.lastEvent:
+                self.lastEvent = event
+                notification(event)
+        except Exception as e: log("checkEvents, Failed " + str(e) + ", event = " + str(event), xbmc.LOGERROR)
+           
+           
+    def chkDeviceStatus(self):
+        deviceInfo = list(self.getDeviceInfo(self.deviceLST))
+        for device in deviceInfo:
+            battery     = (device.get('Battery',None) or None)
+            temperature = (device.get('Temperature',None) or None)
+            if MON_TEMP and temperature is not None:
+                if (temperature > TEMP_MAX or temperature < TEMP_MIN): notification('%s Temperature is %f'%(device['name'],temperature))
+            if MON_BATT and battery is not None:
+                if (battery) <= MON_BATT: notification('%s Battery is %d'%(device['name'],int(battery)))
+        self.nextCHK = datetime.datetime.now() + datetime.timedelta(seconds=self.chkIntval)
+        log('chkDeviceStatus, nextCHK = ' + str(self.nextCHK))
+        return True
+                
+              
+    def startService(self):
+        log('startService')
+        self.myMonitor = Monitor()
+        while not self.myMonitor.abortRequested():
+            if self.myMonitor.pendingChange == True or self.myMonitor.waitForAbort(5) == True: 
+                log('startService, waitForAbort/pendingChange')
+                break
+      
+            # Don't run while playing.
+            if xbmc.Player().isPlayingVideo() and IGNORE:
+                log('startService, ignore during playback')
+                continue
+                
+            # Don't run while setting menu is opened.
+            if xbmcgui.getCurrentWindowDialogId() in [10140,10103,12000]:
+                log('startService, settings dialog opened')
+                continue 
+                
+            self.checkEvents()
+            if datetime.datetime.now() >= self.nextCHK:
+                self.chkDeviceStatus()
+                
+        if self.myMonitor.pendingChange == True:
+            notification(LANGUAGE(30007))
+            self.restartService()
+            
+        
+    def restartService(self):
+        log('restartService')
+        #adapted from advised method https://forum.kodi.tv/showthread.php?tid=248758
+        xbmc.sleep(500)
+        xbmc.executeJSONRPC('{"jsonrpc": "2.0", "method":"Addons.SetAddonEnabled","params":{"addonid":"%s","enabled":false}, "id": 1}'%(ADDON_ID))
```

Comment:
this is a bit hackish and I am not sure it'll work without issues. You're disabling the addon from the the service it is running. Can't you use an alternative option like spanning a new thread and defining a class variable which kills the main service loop?

Reply:
This won't work, Kodi will not release a services settings until it is killed. I've tried multiple ways of killing a service, reloading changed settings and restarting the service. None have worked... The service only detects the settings loaded initially.